lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1175,71 +1242,64 @@ def compute_severity(resource_type, record, entry_common_labels)
       end
     end
 
-    NANOS_IN_A_SECOND = 1000 * 1000 * 1000
+    def set_log_entry_fields(record, entry)
+      LOG_ENTRY_FIELDS_MAP.each do |field_name, config|
+        payload_key, subfields, grpc_class, non_grpc_class = config
+        begin
+          payload_key = instance_variable_get(payload_key)
+          fields = record[payload_key]
+          next unless fields.is_a?(Hash)
+
+          extracted_subfields = subfields.each_with_object({}) \
+            do |(original_key, destination_key, cast_fn), extracted_fields|
+            value = fields.delete(original_key)
+            next if value.nil?
+            begin
+              casted_value = send(cast_fn, value)
+            rescue TypeError
+              log = <<-eos
```

Comment:
Seems `eos` will print the extra spaces and new lines.

```
$ cat test.rb
              log = <<-eos
                Failed to cast_fn for field_name.original_key
                with value value.inspect.
              eos
puts log
$ ruby test.rb
                Failed to cast_fn for field_name.original_key
                with value value.inspect.
```
We could instead try:
```
              @log.error "Failed to #{cast_fn} for #{field_name}." \
                         "#{original_key} with value #{value.inspect}.", err
```